src/deadline/job_attachments/download.py

```diff
@@ -999,7 +1052,9 @@ def _merge_asset_manifests_sorted_asc_by_last_modified(
         return None
 
     # Sort manifests by timestamp (oldest first)
-    sorted_manifests_with_timestamps = sorted(manifests_with_last_modified_timestamps)
```

Comment:
Bugfix: if timestamps are the same, we'd try to compare `BaseAssetManifest` but this type doesn't support comparison.